data_steward/vocabulary.py

```diff
@@ -1,11 +1,179 @@
+
+csv.field_size_limit(sys.maxsize)
+
+DELIMITER = '\t'
+LINE_TERMINATOR = '\n'
+RAW_DATE_REGEX = r'\d{8}$'  # yyyymmdd
+BQ_DATE_REGEX = r'\d{4}-\d{2}-\d{2}$'  # yyyy-mm-dd
+
+TRANSFORM_FILES = 'transform_files'
+APPEND_VOCABULARY = 'append_vocabulary'
+APPEND_CONCEPTS = 'append_concepts'
+ADD_AOU_GENERAL = 'add_aou_general'
+ERRORS = 'errors'
+
+AOU_GEN = 'AoU_General'
+AOU_GEN_VOCABULARY_CONCEPT_ID = '2000000000'
+AOU_GEN_VOCABULARY_REFERENCE = 'https://docs.google.com/document/d/10Gji9VW5-RTysM-yAbRa77rXqVfDfO2li2U4LxUQH9g'
+OMOP_VOCABULARY_CONCEPT_ID = '44819096'
+
+ERROR_APPENDING = 'Appending to {in_path} which already contains rows for ' + AOU_GEN
+
+
+def format_date_str(s):
+    """
+    Format a date string to yyyymmdd if it is not already
+    :param s: the date string
+    :return: the formatted date string
+    """
+    if re.match(BQ_DATE_REGEX, s):
+        return s
+    elif re.match(RAW_DATE_REGEX, s):
+        parts = s[0:4], s[4:6], s[6:8]
+        return '-'.join(parts)
+    else:
+        raise ValueError('Cannot parse value {v} as date'.format(v=s))
+
+
+def _transform_csv(in_fp, out_fp, err_fp=None):
+    if not err_fp:
+        err_fp = sys.stderr
+    csv_reader = csv.reader(in_fp, delimiter=DELIMITER)
+    header = next(csv_reader)
+    date_indexes = []
+    for i in range(0, len(header)):
+        if header[i].endswith('_date'):
+            date_indexes.append(i)
+    csv_writer = csv.writer(out_fp, delimiter=DELIMITER, lineterminator=LINE_TERMINATOR)
+    csv_writer.writerow(header)
+    for row_index, row in enumerate(csv_reader):
+        try:
+            for i in date_indexes:
+                row[i] = format_date_str(row[i])
+            csv_writer.writerow(row)
+        except Exception, e:
+            message = 'Error %s transforming row:\n%s' % (e.message, row)
+            err_fp.write(message)
+
+
+def transform_file(file_path, out_dir):
+    """
+    Format file date fields and standardize line endings a local csv file and save result in specified directory
+
+    :param file_path: Path to the csv file
+    :param out_dir: Directory to save the transformed file
+    """
+    file_name = os.path.basename(file_path)
+    table_name, _ = os.path.splitext(file_name.lower())
+    out_file_name = os.path.join(out_dir, file_name)
+    err_dir = os.path.join(out_dir, ERRORS)
+    err_file_name = os.path.join(err_dir, file_name)
+    if not os.path.exists(err_dir):
+        os.makedirs(err_dir)
+    with open(file_path, 'rb') as in_fp, open(out_file_name, 'wb') as out_fp, open(err_file_name, 'wb') as err_fp:
+        _transform_csv(in_fp, out_fp, err_fp)
+
+
+def transform_files(in_dir, out_dir):
+    """
+    Transform vocabulary files in a directory and save result in another directory
+
+    :param in_dir: Directory containing vocabulary csv files
+    :param out_dir: Directory to save the transformed file
+    """
+    fs = os.listdir(in_dir)
```

Comment:
I suggest changing `f` and `fs` to longer variable names to make them support self documentation.